javaagent-instrumentation-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/InstrumentationContext.java

```diff
@@ -28,7 +33,27 @@ private InstrumentationContext() {}
    */
   public static <Q extends K, K, C> ContextStore<Q, C> get(
       Class<K> keyClass, Class<C> contextClass) {
-    throw new IllegalStateException(
-        "Calls to this method will be rewritten by Instrumentation Context Provider (e.g. FieldBackedProvider)");
+    if (contextStoreSupplier == null) {
+      throw new IllegalStateException("");
+    }
+    return contextStoreSupplier.get(keyClass, contextClass);
```

Comment:
I wonder if we still need to replace `InstrumentationContext.get()` calls in `FieldBackedProvider` - if the provider idea works for helper classes, wouldn't it work everywhere? Removing that part would certainly make `FieldBackedProvider` a bit simpler.

Reply:
I believe that it is better to keep replacing `InstrumentationContext.get()` calls in `FieldBackedProvider` because this `InstrumentationContext.get()` implementation isn't really efficient. It involves `Class.forName` and `Class.getMethod` which is fine in helper classes where you can do the lookup once and keep `ContextStore` in a field.